Makefile

```
@@ -2,15 +2,56 @@ LIBRARY_NAME=libpubliccode-parser
 
 LIB_DIR=lib
 
-LIBRARY=$(LIB_DIR)/$(LIBRARY_NAME).so
+# Detect current platform
+GOOS := $(shell go env GOOS)
+GOARCH := $(shell go env GOARCH)
 
-.PHONY: all build clean deps
+# Determine file extension based on OS
+ifeq ($(GOOS),darwin)
+	EXT=dylib
+else
+	EXT=so
+endif
+
+# Current platform library
+LIBRARY=$(LIB_DIR)/$(LIBRARY_NAME)-$(GOOS)-$(GOARCH).$(EXT)
+
+# Legacy symlink for backward compatibility
+LEGACY_LIBRARY=$(LIB_DIR)/$(LIBRARY_NAME).so
+
+.PHONY: all build build-all build-linux-amd64 build-linux-arm64 build-darwin-amd64 build-darwin-arm64 clean install
```

```suggestion
.PHONY: all build build-all build-linux-amd64 build-linux-arm64 build-darwin-arm64 clean install
```

Some of these are not phony targets, but we can fix that later